Change wording on toServices limitations (see #20067)

atykhyy opened the pull request:
#21052 updated Cilium documentation to say that, in network policy rules, `toServices` statements _cannot_ be combined with `toPorts` statements. I believe it would be more informative for users of Cilium to say (following RFC 2119) that `toServices` _must not_ be combined with `toPorts`, as technically Cilium accepts such a configuration as valid but handles it in the unexpected and unuseful manner described in #20067.

qmonnet reviewed (approved):
Looks good, thank you!

christarazi reviewed (commented):
Thanks for the PR!

Checkpatch (checks your commits) is failing with 

```
   Warning: WARNING:TYPO_SPELLING: 'unuseful' may be misspelled - perhaps 'useless'?
  #11: 
  as valid but handles it in the unexpected and unuseful manner described
                                                ^^^^^^^^
```

atykhyy committed:
Change wording on toServices limitations (see #20067)

PR #21052 updated Cilium documentation to say that, in network policy
rules, `toServices` statements cannot be combined with `toPorts`
statements. I believe it would be more informative for Cilium users
to say (following RFC 2119) that `toServices` _must not_ be combined
with `toPorts`, as technically Cilium accepts such a network policy
as valid but handles it in the unexpected and potentially dangerous
(e.g. if a setup relies on Cilium network policy to implement egress
filtering) manner described in #20067.

Signed-off-by: Anton Tykhyy <atykhyy@gmail.com>

atykhyy committed:
Make rules combining toServices and toPorts fail validation

Currently, an egress rule combining toServices and toPorts has
the unexpected and potentially dangerous side effect of allowing egress
traffic to any remote endpoint on the port(s) specified (see #20067).
This change makes such rules fail validation.

Signed-off-by: Anton Tykhyy <atykhyy@gmail.com>
